From 64d07df506d1b42c7542e56d4c70e1afb3172b86 Mon Sep 17 00:00:00 2001
From: Yuan Juntao <juntao.yuan@intel.com>
Date: Tue, 3 Feb 2015 09:12:05 +0800
Subject: [PATCH] mmc: core: optimize using of mmc_delayed_work_wake_lock

Release this wake lock correctly to prevent from blocking system D3
procedure.

Change-Id: I840a11b8afd2ae34f967c93237e734c4b3998306
Signed-off-by: Yuan Juntao <juntao.yuan@intel.com>
---
 drivers/mmc/core/core.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/core/core.c b/drivers/mmc/core/core.c
index 506383e..c65cfcc 100644
--- a/drivers/mmc/core/core.c
+++ b/drivers/mmc/core/core.c
@@ -2557,11 +2557,11 @@ void mmc_rescan(struct work_struct *work)
 	bool extend_wakelock = false;
 
 	if (host->rescan_disable)
-		return;
+		goto out;
 
 	/* If there is a non-removable card registered, only scan once */
 	if ((host->caps & MMC_CAP_NONREMOVABLE) && host->rescan_entered)
-		return;
+		goto out;
 	host->rescan_entered = 1;
 
 	mmc_bus_get(host);
-- 
1.9.1

